src/main/kotlin/io/bitvavo/simulation/exchange/SimulatorExchange.kt

```diff
@@ -0,0 +1,40 @@
+package io.bitvavo.simulation.exchange
+
+import io.bitvavo.simulation.engine.MatchingEngine
+import io.bitvavo.simulation.exchange.mappers.OrderMapper.tryParseOrder
+import io.bitvavo.simulation.exchange.sequence_generator.SequenceNumberGenerator
+import io.bitvavo.simulation.models.OrderBookSnapshot
+import io.bitvavo.simulation.models.OrderMatchResult
+import io.bitvavo.simulation.models.Trade
+
+class SimulatorExchange(private val engine: MatchingEngine, private val generator: SequenceNumberGenerator) : Exchange {
+    override var onTrade: ((trades: List<Trade>) -> Unit)? = null
+    override var onFailure: ((reason: String) -> Unit)? = null
+    override var onSnapshot: ((orderBook: OrderBookSnapshot) -> Unit)? = null
+
+    override fun executeCommand(cmd: String) {
```

Comment:
Maybe we can parse the cmd before passing it to Exchange? So that it doesn't know about anything except orders?